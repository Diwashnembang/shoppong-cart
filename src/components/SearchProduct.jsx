import SearchIcon from '@mui/icons-material/Search';
import CloseIcon from '@mui/icons-material/Close';

import React from 'react'
import "./search.css"

const SearchProduct = (props) => {
    const { showSearch } = props;
    const handleCloseClick = () => {
        showSearch(false);

    }
    return (
        <div id='search'>
            <section className='close' onClick={handleCloseClick}><CloseIcon fontSize='large' /></section>

            <section className="input">
                <div className="search-logo">
                    <SearchIcon fontSize='16' />
                </div>
                <input type="text" name="search" id="product-search" placeholder='Search product' />

            </section>

        </div>
    )
}

export default SearchProduct