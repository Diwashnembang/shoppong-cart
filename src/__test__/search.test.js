
import React from "react"
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"
import SearchProduct from "../components/SearchProduct"
import userEvent from "@testing-library/user-event"

const mockShowSearch = jest.fn();
describe("SearchProduct Component", () => {
    it("renders correctly", () => {
        const { container } = render(<SearchProduct />);
        expect(container).toMatchSnapshot();
    })

    it("should call showSearch with false value when close button is clicked", () => {
        render(<SearchProduct showSearch={mockShowSearch} />)

        const closeIconElement = screen.getByTestId("close");
        userEvent.click(closeIconElement);
        expect(mockShowSearch).toBeCalledWith(false);
    })

})