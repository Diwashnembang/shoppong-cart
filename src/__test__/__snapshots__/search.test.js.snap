// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProduct Component renders correctly 1`] = `
<div>
  <div
    id="search"
  >
    <section
      class="close"
      data-testid="close"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </section>
    <section
      class="input"
    >
      <div
        class="search-logo"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSize16 css-821wyw-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </div>
      <input
        id="product-search"
        name="search"
        placeholder="Search product"
        type="text"
      />
    </section>
  </div>
</div>
`;
